### PR TITLE
Basic config

### PR DESCRIPTION
**Pull request by aartyszko-stx**
eslint, prettier, husky, mochawsome configuration has been added

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "env": {
+    "browser": true,
+    "es2021": true
+  },
+  "extends": [
+    "airbnb-base",
```
Comment (bbrozyna) [inline .eslintrc.json]: what's that? :D

File: package.json [hunk 2]
```diff
@@ -4,11 +4,26 @@
   "description": "",
   "main": "cypress.config.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "cy:open": "cypress open",
```
Comment (bbrozyna) [inline package.json]: --e2e maybe

File: package.json [hunk 3]
```diff
@@ -4,11 +4,26 @@
...
+    "cy:run": "cypress run",
+    "lint": "eslint .",
```
Comment (bbrozyna) [inline package.json]: It means pre-commit will format files using prettier but only check them using eslint?

File: package.json [hunk 4]
```diff
@@ -4,11 +4,26 @@
...
+    "lint:fix": "eslint --fix .",
```
Comment (bbrozyna) [inline package.json:10]: I stole it from cypress-reaworld-app, what do you think?
    "lint": "eslint && prettier --check \"**/**.{ts,js,tsx}\" \"*.{json,md,yml}\"",

File: package.json [hunk 5]
```diff
@@ -4,11 +4,26 @@
...
+    "prepare": "husky install",
+    "prettier": "prettier --write \"**/*.{js,json,ts}\""
```
Comment (bbrozyna) [inline package.json]: "prettier": "prettier --write '**/**.{ts,js,tsx}' '*.{json,md,yml}'", what about yamls and readme! :D

File: package.json [hunk 6]
```diff
@@ -4,11 +4,26 @@
...
   },
   "author": "STX Next",
   "license": "ISC",
   "devDependencies": {
-    "cypress": "^11.0.0"
+    "@typescript-eslint/eslint-plugin": "^6.3.0",
+    "@typescript-eslint/parser": "^6.3.0",
+    "cypress": "^11.0.0",
+    "cypress-mochawesome-reporter": "^3.5.1",
```
Comment (bbrozyna) [inline package.json:20]: It works :3

File: prettier.config.json [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  "useTabs": false,
+  "tabWidth": 2,
+  "bracketSpacing": true,
+  "arrowParens": "avoid",
+  "trailingComma": "all",
+  "preferSingleLine": false,
+  "memberExpression": {
+    "linePerExpression": true
+  },
+  "printWidth": 140,
```
Comment (bbrozyna) [inline prettier.config.json:14]: We go wide!

Comment (bbrozyna) [reply prettier.config.json:14]: I respect that config with hate on single quotes, trailing comas and "avoid" mode :)

File: .prettierignore [hunk 8]
```diff
@@ -0,0 +1 @@
+reports
```
Comment (bbrozyna) [inline .prettierignore]: Prettier without empty lines on its own file :P ironic

Comment (bbrozyna) [inline .prettierignore]: Mochawesome reports aren't prettier compliant? 😨

File: package.json (hunk 3 above)
Comment (aartyszko-stx) [reply package.json]: yup, do you think it would be better to run eslint:fix instead?

Comment (aartyszko-stx) [reply package.json]: Nevermind, I saw comment below

File: package.json (hunk 4 above)
Comment (aartyszko-stx) [reply package.json:10]: Something was no yes with this command, I modified it a bit so now it should be fine

**Review by bbrozyna — APPROVED** [review]
Yolo

**Pull request merged**